source/_posts/2016-11-08-Vapor-setup.md

```diff
@@ -0,0 +1,311 @@
+	id: null,
+	milesDriven: 1000,
+	name: "Fiat"
+}
+```
+The id will still be null because the object wasn't instanciet from the db.
+
+### 6. Adding some more magic
+
+Ok cool, let's make our API accutally hold some data (remember this will only be held untill you have run the project again. Open your ```CarController.swift``` file and update the code to this:
+
+```swift
+import Vapor
+import HTTP
+
+final class CarController: ResourceRepresentable {
+    
+    //Adding our session container object
+    var cars:[Car] = []
+    
+    func index(request: Request) throws -> ResponseRepresentable {
+        
+        //Update method to return the container object instead of the static object
+        return try JSON(node: cars)
+    }
+
+    //This is where the 'post' request gets redirected to
+    func create(request: Request) throws -> ResponseRepresentable {
+        
+        //Guard statement to make sure we are validating the data correct (we ofc should also later quard for the color etc)
+        guard let name = request.data["name"]?.string else {
+            //Throw a Abort response, I like using the custom status to make sure the frontends have the correct message and response code
+            throw Abort.custom(status: Status.preconditionFailed, message: "Missing name")
+        }
+        
+        //Create a car
+        let car = Car(name: name, color: "Red", milesDriven: 0)
+        //Add it to our container object
+        cars.append(car)
+        //Return the newly created car
+        return try car.converted(to: JSON.self)
+    }
+    
+    //Add the store: create to tell Vapor that if a 'post' http requst comes in to redirect it there.
+    func makeResource() -> Resource<Car> {
+        return Resource(
+            index: index,
+            store: create
+        )
+    } 
+}
+```
+Now Run your Xcode project again. If you start by opening your browser at our 'get' endpoint ```http://0.0.0.0:8080/api/cars``` for the cars then you will ofc be able to see that it's returning an empty array.
+
+So let's now try to make a post request to our 'post endpoint' witch is basicly the same slug ```http://0.0.0.0:8080/api/cars``` (you can use whatever API testing tool, I normally uses [Postman](https://www.getpostman.com/). If you don't have any tools like that then just use this cURL:
+
+```jconsole
+curl -H "Content-Type: application/json" -X POST -d '{}' http://0.0.0.0:8080/api/cars
+```
+
+This should return our error response (response code 412):
+```javascript
+{"error":true,"message":"Missing name"}
+```
+
+Let's now add our name param to the cURL:
+
+```jconsole
+curl -H "Content-Type: application/json" -X POST -d '{"name":"Fiat"}' http://0.0.0.0:8080/api/cars
+```
+
+And volia! we have stored our first (only in session object (and should ofc see our stored car object as JSON response). Try running the curl (or postman action) a couple of times with different names, and then  run the 'get' in the browser..
```

Comment:
s/volia/voila
There are 2 whitespaces in front of 'run'. Remove one of them
s/browser../browser.     (one dot after browser)